mmdeploy/codebase/mmseg/models/segmentors/encoder_decoder.py

```diff
@@ -25,16 +23,7 @@ def encoder_decoder__simple_test(ctx, self, img, img_meta, **kwargs):
         torch.Tensor: Output segmentation map pf shape [N, 1, H, W].
     """
     seg_logit = self.encode_decode(img, img_meta)
-    seg_logit = resize(
-        input=seg_logit,
-        size=img_meta['img_shape'],
-        mode='bilinear',
-        align_corners=self.align_corners)
     seg_logit = F.softmax(seg_logit, dim=1)
-    seg_pred = seg_logit.argmax(dim=1)
-    # our inference backend only support 4D output
-    shape = seg_pred.shape
-    if not is_dynamic_shape(ctx.cfg):
-        shape = [int(_) for _ in shape]
-    seg_pred = seg_pred.view(shape[0], 1, shape[1], shape[2])
+    # keep 4D output required by backends such as TensorRT
```

Comment:
It is not required by any backends.
We do a nearest resize outside the model in `segmentation_model` as the postprocess, which require 4d input.

Reply:
It seems that trt requires 4d output as tested before when supporting deployment feature in mmseg

Reply:
It is caused by postprocessing even in mmseg. We can just remove this comment.